llvm/lib/Target/PowerPC/PPCISelLowering.cpp

```diff
@@ -8080,7 +8080,8 @@ SDValue PPCTargetLowering::LowerSELECT_CC(SDValue Op, SelectionDAG &DAG) const {
   // For more information, see section F.3 of the 2.06 ISA specification.
   // With ISA 3.0
   if ((!DAG.getTarget().Options.NoInfsFPMath && !Flags.hasNoInfs()) ||
-      (!DAG.getTarget().Options.NoNaNsFPMath && !Flags.hasNoNaNs()))
+      (!DAG.getTarget().Options.NoNaNsFPMath && !Flags.hasNoNaNs()) ||
+      ResVT == MVT::f128)
```

Comment:
does that mean fsel also does not support `ppc_fp128`?

Reply:
yes, we don't have instruction for `ppc_fp128` fsel. The motivating case exists from https://github.com/llvm/llvm-project/commit/097a95f2df4666df74909ab1d3bcc0c1824ace86 , which sets `select_cc f128` as custom, while we don't have that lowering for `ppc_fp128`.

Reply:
But https://github.com/llvm/llvm-project/commit/097a95f2df4666df74909ab1d3bcc0c1824ace86 will not try to custom `SELECT_CC` to fsel, right? That commit is only for ieee long double.
Do we have other pattern that will generate fsel operating on ppc_fp128 type?

Reply:
No. `PPCISD::FSEL` is only generated in `LowerSELECT_CC`, while `SELECT_CC ppc_fp128` is expand.

Reply:
Thank-you!